File: rb/features.rb
# TelegramBot SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module TelegramBotFeatures
  def self.make_feature(name)
    case name
    when "base"
      TelegramBotBaseFeature.new
    when "test"
      TelegramBotTestFeature.new
    else
      TelegramBotBaseFeature.new
    end
  end
end

File: py/features.py
# TelegramBot SDK feature factory

from feature.base_feature import TelegramBotBaseFeature
from feature.test_feature import TelegramBotTestFeature


def _make_feature(name):
    features = {
        "base": lambda: TelegramBotBaseFeature(),
        "test": lambda: TelegramBotTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
